File: 백준/Silver/1449. 수리공 항승/수리공 항승.py
import sys
input = sys.stdin.readline

N, L = map(int, input().split())
p = list(map(int, input().split()))
p.sort()

result = 0
start = p[0]

for i in range(1, len(p)):
    if (p[i] - start) + 1 > L:
        result += 1
        start = p[i]
result += 1
print(result)

File: 백준/Silver/1449. 수리공 항승/수리공 항승.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N = Integer.parseInt(st.nextToken());
        int L = Integer.parseInt(st.nextToken());
        
        int[] p = new int[N];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            p[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(p);
        
        int result = 0;
        int start = p[0];

        for(int i = 1; i < p.length; i++) {
            if(p[i] - start + 1 > L) {
                result++;
                start = p[i];
            }
        }
        
        result++;
        
        System.out.print(result);
    }
}